### PR TITLE
Changes to make person editable (resolves #89)

### LINKED ISSUE #89: Make Person properties editable on the frontend

### PR DESCRIPTION
**Pull request by cr33dx**
## Changes

Make the person's table data editable(Front-end)

Screenshot 2020-08-24 at 1 14 33 PM

## Checklist

- [x] All querysets/queries filter by Team (if this PR affects any querysets/queries)
- [x] Backend tests (if this PR affects the backend)
- [x] Cypress E2E tests (if this PR affects the front and/or backend)

### CODE REVIEW

**Comment by yakkomajuri** [comment]
Hey @cr33dx! First of all, thanks for this! I know how you had some trouble setting up a local instance over the weekend so I appreciate you taking the time. 

However, I believe this doesn't address the issue in full. I think the idea is that one could edit the properties on the frontend but have those changes be reflected in the backend and saved in the database. I think perhaps the name of the issue could have led to some confusion, but the idea is that the same PR makes it so that the properties can be edited on the PostHog UI by making changes to both our frontend and backend. I believe the approach here is to establish a new endpoint for the API that allows properties to be changed for a given user.

**Review by yakkomajuri — CHANGES_REQUESTED** [review]
Backend changes needed

**Comment by cr33dx** [comment]
Got it @yakkomajuri

**Comment by yakkomajuri** [comment]
If you have any questions feel free to send them over!

**Comment by cr33dx** [comment]
How does the front end code look?

**Comment by yakkomajuri** [comment]
Seems ok, but I haven't looked in too deep, would rather put out a full review once this is all done!

**Comment by Twixes** [comment]
Happy to review as soon as this covers the backend in addition to the frontend!

**Comment by timgl** [comment]
@cr33dx Hey, did you get a chance to take another look at the backend for this?

**Comment by cr33dx** [comment]
@timgl was busy over the week will work on it this weekend plus I’m new to django so took a while to get around the concepts.

**Comment by timgl** [comment]
No worries at all! Let me know when you get stuck or need any help :)

**Comment by cr33dx** [comment]
Hi Team apologies for the delay not getting time to look at the backend cause of work at job.

**Comment by timgl** [comment]
@cr33dx I just double checked, there's no need to make any backend changes. All you need to do is send a PATCH request to `api/person/:person_id` with the new properties. That should work!

**Comment by cr33dx** [comment]
don't know why the cypress tests are failing things are working on my end when i checked few of the failed checks.
one of them was 
`data-attr=funnel-viz is expected to be on DOM` 
and it was there

Screenshot 2020-09-08 at 1 15 58 AM

**Comment by yakkomajuri** [comment]
Cypress tests might not have anything to do with you @cr33dx - I'll aim to review this today or tomorrow

**Comment by yakkomajuri** [comment]
Hey @cr33dx this isn't working just yet - You need to pass the data to be updated along with your PATCH request. 

Another comment is that I'm not a big fan of the browser alert. We should probably substitute that for a little modal.

**Comment by cr33dx** [comment]
hi @yakkomajuri i'm passing PATCH req with the new data and getting 200 on backend. i'll replace the front end with the modal though planning to make a saveModal modal on scenes/modals/saveModal.js where we can pass save function as props.

File: frontend/src/lib/utils.js [hunk 1]
```diff
@@ -241,6 +241,13 @@ export const deletePersonData = (person, callback) => {
         })
 }
 
+export const savePersonData = (person) => {
+    window.confirm("save person's data") &&
+        api.update('api/person/' + person.id).then(() => {
```
Comment (yakkomajuri) [inline frontend/src/lib/utils.js:246]: @cr33dx You will get a 200 on the PATCH request even if you do not send any data. Here, `update` takes `data` as a second parameter

**Comment by yakkomajuri** [comment]
To make sure this is working, you should refresh the page after making the changes and see if they persist

**Comment by cr33dx** [comment]
@yakkomajuri really sorry did not saw that the latest commit is updating the person's data in backend.

**Comment by cr33dx** [comment]
Screenshot 2020-09-10 at 1 40 00 AM

**Review by yakkomajuri — CHANGES_REQUESTED** [review]
Thanks @cr33dx! This is great, but there are still some issues here:

1. I'm unable to load the profiles of people who only have the `is_demo` property and nothing else
2. I'm unable to edit `is_demo`
3. The button for saving the person's data does not work for me when I change the screen's dimensions with DevTools. The delete button right above it works just fine.
4. The system for checking if the person's details were updated is not "smart" - this _isn't a big deal_ but it would be nice if it knew that I changed and then went back, rather than just setting `personChanged` blindly on every change. 
5. Also not a _requirement_, but I feel like I would prefer clicking a button to toggle on editing rather than being able to edit by default. No need to worry about this before points 1-3 are solved, however.

**Comment by cr33dx** [comment]
Hi, i did not get the first point.

**Review by yakkomajuri — CHANGES_REQUESTED** [review]
@cr33dx thanks once more! This is almost there and I'm so excited for this feature. I put out some minor fixes to wording and style (mostly responsiveness-related), but there's one thing that's still going wrong: We can now handle strings and booleans as properties, but numbers for example don't work. You can refer to [this comment and PR](https://github.com/PostHog/posthog/pull/1666#issuecomment-694030031) for the types of the properties we support. Once we make sure all those properties are correctly handled, this should be good to go!

**Comment by yakkomajuri** [comment]
Note to self: 

**On new PRs** implement adding a property, deleting a property, and update the delete person confirm to a modal.

**Comment by cr33dx** [comment]
- If a property is left empty by user we mark it as null.
- Using this (object)[https://github.com/PostHog/posthog/blob/master/frontend/src/lib/components/PropertyKeyInfo.js], disable posthog's internal key value pair.

**Comment by cr33dx** [comment]
hi guys any update?

**Comment by Twixes** [comment]
Yep, will take a look in a moment!

**Review by Twixes — APPROVED** [review]
A bit rough around the edges, but it's pretty practical and we can always improve it.

**Pull request merged**